Update environment-setup (#2081)
Seneca-CDOT/telescope: docs/environment-setup.md

## After installing the prerequisites:

### Start Docker

```
sudo systemctl start docker
```

_Note: You may need to add your user to the docker group in Linux to use `docker-compose` without `sudo`. To do this, try `groups $USER` in a terminal and check if docker is in the list of groups. If not, add it with `usermod -aG docker $USER` and reboot._

**Important:** Docker builds Telescope's dependencies at launch and keeps them on disk. In some cases, Docker might try to reuse already-built dependencies or cached data, causing misleading results when testing Telescope. To avoid this, it is recommended to use the command `docker system prune -af --volumes` to remove all already-built Telescope dependencies and ensure fresh deployments.
More information about docker: [images vs containers](https://www.baeldung.com/docker-images-vs-containers) and [volumes](https://docs.docker.com/storage/volumes/).

### Start telescope

> There are different ways to run the application. By default, `.env` will be read; however, if there is no `.env` file in the root, [env.development](../config/env.development) will be used.

Reviewer: I think the `../config/env.development` link won't work, will it? Should we use the link to the master branch instead or just add the path as text, i.e. just `../config/env.development`?

Reviewer 2: What @yuanLeeMidori has is right, it works this way.

Reviewer 2: Saying that `.env` will be used is a bit misleading.  We only use `.env` for the old back-end, and for the way the front-end reads a few URLs. We're trying to remove the need for an `.env` file, but still support it in some limited ways (e.g. for frontend overrides).

> There are also [env.production](../config/env.production) and [env.staging](../config/env.staging) to choose based on developer's need. For example, if you want to use staging, you can do `cp config/env.staging .env` on Linux/macOS, or `copy config/env.staging .env` on windows.

Reviewer: Same thing with the links here

Here are instructions for different scenarios:
_Note: Make sure you're running these commands in the root of telescope project. If any of the commands below are failing, use the command `pwd` to find your current directory and navigate back to project root (e.g., `cd <the path of directory you place telescope project>/telescope`)_

#### Want to run back-end services locally, and to have front-end using the services

This is the default setting, you do not need to copy or modify any `env` file.

> ```

Reviewer: Should we add `bash` after the opening '```' to support highlighting?

Author: yes, thank you!

npm run services:start

npm run dev
```

Back-end will start downloading feeds and processing them until stopped. The default port # for the back-end is `3000` and can be modified with the `PORT` variable in the `.env` file. Different routes such as `http://localhost:3000/posts` and
`http://localhost:3000/feeds` now return data. And front-end can be found in `http://localhost:8000` by default.

If this doesn't work for you, it is possible that you have an old `.env` file in the root that you copied from `env.example` from telescope 1.0. Please remove it, and try again.

#### Working strictly on the front-end

> ```

Reviewer: Should we add `bash` after the opening '```' to support highlighting?

> cp config/env.staging .env

Reviewer: Maybe use strict path: `cp ./config/env.staging ./.env`?

npm run dev
```

> This Will provide you staging back-end without running it locally.

Reviewer 2: `Will -> will`

#### Want to mix and match services between local and staging

> See [staging-production-deployment](staging-production-deployment) for more information on running Telescope in staging or production mode.

Reviewer: Should this be a link to an actual file?

> This one depends on which part you're working with. For example, if you want to work with authorization, you need to specify the URL of AUTH in your `.env` file, which is, go to `.env` and find `AUTH_URL=...` and modify it to the one you want to work with. If you're testing auth locally, use `AUTH_URL=http://localhost/v1/auth`; otherwise, use the staging one, `AUTH_URL=http://dev.api.telescope.cdot.systems/v1/auth`.

Reviewer: ```suggestion
This one depends on which part you're working with. For example, if you want to work with authorization, you need to specify the URL of AUTH in your `.env` file by going to `.env` and modifying `AUTH_URL=...` to the one you want to work with. If you're testing auth locally, use `AUTH_URL=http://localhost/v1/auth`; otherwise, use the staging one, `AUTH_URL=http://dev.api.telescope.cdot.systems/v1/auth`.
```

> After modify the `.env` file, run these commands,

Reviewer: ```suggestion
After modifying the `.env` file, run the following commands:
```

> ```

Reviewer: Should we add bash after the opening '```' to support highlighting?

npm run services:start

npm run dev
```

> #### Want to run auth/image/posts service solely

Reviewer: ```suggestion
#### Want to run `auth/image/posts` service solely
```

`npm run services:start auth` or `npm run services:start image` or `npm run services:start posts`

> #### After making changes and would like to make Docker image reflects

Reviewer 2: "...would like to update the Docker image(s)"

> #### After making changes and would like to update the Docker image(s)

Reviewer: ```suggestion
#### Want to update the Docker image(s) after making some changes
```

> `npm run services:clean` and `npm run services:start` to rebuild the image.

Reviewer: ```suggestion
Run the following commands to rebuild the image:
```bash
npm run services:clean
npm run services:start
```
```

#### Login/SSO:

> If you need to login to Telescope or developing/testing requiring a logged in user, you don't need to start an extra container for login, it is included in auth service. You can simply use UI to login. For more information on Login please refer to our [Login Document](login.md).

Reviewer: ```suggestion
If you need to login to Telescope or your work requires logging in to be tested, you don't need to start an extra container for login, it is included in auth service. You can simply use UI to login. For more information on Login please refer to our [Login Document](login.md).
```

> If you need to login to Telescope or developing/testing requiring a logged in user, you don't need to start an extra container for login, it is included in auth service. You can simply use UI to login. For more information on Login please refer to our [Login Document](login.md).

Reviewer: Should `[Login Document](login.md)` be a link to an actual file on GitHub?

### Start telescope

There are different ways to run the application. By default, `.env` will be read; however, if there is no `.env` file in the root, [env.development](../config/env.development) will be used.

There are also [env.production](../config/env.production) and [env.staging](../config/env.staging) to choose based on developer's need. For example, if you want to use staging, you can do `cp ./config/env.staging ./.env` on Linux/macOS, or `copy config/env.staging .env` on windows.

Here are instructions for different scenarios:
_Note: Make sure you're running these commands in the root of telescope project. If any of the commands below are failing, use the command `pwd` to find your current directory and navigate back to project root (e.g., `cd <the path of directory you place telescope project>/telescope`)_

#### Want to run back-end services locally, and to have front-end using the services

This is the default setting, you do not need to copy or modify any `env` file.

```bash
npm run services:start

npm run dev
```

> Back-end will start downloading feeds and processing them until stopped. The default port # for the back-end is `3000` and can be modified with the `PORT` variable in the `.env` file. Different routes such as `http://localhost:3000/posts` and

Reviewer 3: Microservices instead of back-end here. Remove reference to the default ports because the services all use different ports. You can alternatively link to the README.md in the src/api directory

> `http://localhost:3000/feeds` now return data. And front-end can be found in `http://localhost:8000` by default.

Reviewer 3: Remove this
